### PR TITLE
[Merged by Bors] - chore: split `IntegralCurve`

### PR DESCRIPTION
**Pull request by winstonyin**
Split `Mathlib.Geometry.Manifold.IntegralCurve` into three files:
* `Mathlib.Geometry.Manifold.IntegralCurve.Basic`: Definitions and basic lemmas about continuity and derivatives of integral curves
* `Mathlib.Geometry.Manifold.IntegralCurve.Transform`: Lemmas about translation and scaling of the domain of integral curves by a constant
* `Mathlib.Geometry.Manifold.IntegralCurve.ExistUnique`: Local existence and uniqueness theorems for integral curves

No content has been substantively changed from the original file. This is to prepare for #9013, which proves a sufficient condition for global existence of integral curves. That'll go in another new file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/IntegralCurve/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2024 Winston Yin. All rights reserved.
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/IntegralCurve/Basic.lean:2]: Please keep the copyright at the year this contribution was really "made".
```suggestion
Copyright (c) 2023 Winston Yin. All rights reserved.

File: Mathlib/Geometry/Manifold/IntegralCurve/ExistUnique.lean [hunk 2]
```diff
@@ -1,32 +1,15 @@
 /-
-Copyright (c) 2023 Winston Yin. All rights reserved.
+Copyright (c) 2024 Winston Yin. All rights reserved.
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/IntegralCurve/ExistUnique.lean:2]:
```suggestion
Copyright (c) 2023 Winston Yin. All rights reserved.

File: Mathlib/Geometry/Manifold/IntegralCurve/Transform.lean [hunk 3]
```diff
@@ -0,0 +1,144 @@
+/-
+Copyright (c) 2024 Winston Yin. All rights reserved.
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/IntegralCurve/Transform.lean:2]:
```suggestion
Copyright (c) 2023 Winston Yin. All rights reserved.

File: Mathlib/Geometry/Manifold/IntegralCurve/Basic.lean [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+* `IsIntegralCurve γ v`: `γ t` is tangent to `v (γ t)` for all `t : ℝ`. That is, `γ` is a global
+integral curve of `v`.
+* `IsIntegralCurveOn γ v s`: `γ t` is tangent to `v (γ t)` for all `t ∈ s`, where `s : Set ℝ`.
+* `IsIntegralCurveAt γ v t₀`: `γ t` is tangent to `v (γ t)` for all `t` in some open interval
+around `t₀`. That is, `γ` is a local integral curve of `v`.
+
+For `IsIntegralCurveOn γ v s` and `IsIntegralCurveAt γ v t₀`, even though `γ` is defined for all
+time, its value outside of the set `s` or a small interval around `t₀` is irrelevant and considered
+junk.
+
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/IntegralCurve/Basic.lean:27]: Can you mention the other files as well? Either as a reference (this file proves xxx, the other files prove yyy), or even a detailed list. ("The following properties are shown in `xxx`." The module doc-string can be helpful for discovering results, so referring to other files is fine and can be beneficial!)

**Review by grunweg — CHANGES_REQUESTED** [review]
Two small comments, otherwise LGTM. Thanks for doing this!

**Comment by grunweg** [comment]
Thanks!
maintainer merge

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**